### ISSUE TITLE
Webhook causes main container to crash

### ISSUE THREAD

**Issue by Starttoaster** [automatisch/automatisch#812]
I'm setting up a webhook app in Automatisch 0.3.0, but invoking the created webhook URL seems to cause the `main` container to crash. 

I ran `curl https://my.domain/webhooks/4290d3d8-9470-4a53-af43-c63ba876cd73` to cause this.

Here's all relevant logging at the time.
```
2022-12-19 05:47:34:4734 [info]: Server is listening on 443
/usr/local/share/.config/yarn/global/node_modules/bcrypt/bcrypt.js:137
        error = new Error('data and salt arguments required');
                ^

Error: data and salt arguments required
    at Object.hash (/usr/local/share/.config/yarn/global/node_modules/bcrypt/bcrypt.js:137:17)
    at /usr/local/share/.config/yarn/global/node_modules/bcrypt/promises.js:29:12
    at new Promise (<anonymous>)
    at Object.module.exports.promise (/usr/local/share/.config/yarn/global/node_modules/bcrypt/promises.js:20:12)
    at Object.hash (/usr/local/share/.config/yarn/global/node_modules/bcrypt/bcrypt.js:133:25)
    at exports.default (/usr/local/share/.config/yarn/global/node_modules/@automatisch/backend/dist/src/controllers/webhooks/handler.js:53:48)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

The bcrypt error here happens exactly when I call the webhook. Perhaps the issue is because I migrated from 0.2.0 to 0.3.0? Or maybe I'm missing some piece of configuration for webhooks somehow.

**Comment by farukaydin** [comment]
@Starttoaster Could you define the `WEBHOOK_SECRET_KEY` as it's mentioned in the [configuration page](https://automatisch.io/docs/advanced/configuration) and check whether it's still happening or not?

**Comment by Starttoaster** [comment]
> @Starttoaster Could you define the `WEBHOOK_SECRET_KEY` as it's mentioned in the [configuration page](https://automatisch.io/docs/advanced/configuration) and check whether it's still happening or not?

I was under the impression the docker-compose entrypoint script set that environment variable for me, based on what's written in `/automatisch/storage.env`. Is that incorrect?

**Comment by Starttoaster** [comment]
I tried also adding it to the env of the container and the issue persists.

```
/automatisch # env | grep WEBHOOK
WEBHOOK_SECRET_KEY=<redacted>
```

**Comment by barinali** [comment]
@Starttoaster, I believe this only occurs in GET requests where our internal ID calculation causes the problem. If you use other supported HTTP methods, it should be fine.

This is already fixed in #787. It'll be fine as of the next release.

**Comment by Starttoaster** [comment]
It's also  happening for me on PUT and POST requests (as well as other verbs probably.) But I guess I'll check out the next release for that feature.

**Comment by barinali** [comment]
@Starttoaster could you please share a sample cURL of the request to the webhook with me?

**Comment by Starttoaster** [comment]
Yeah. I shared a GET request in the OP. A POST request would be:

```
curl -X POST https://my.domain/webhooks/4290d3d8-9470-4a53-af43-c63ba876cd73
```

**Comment by barinali** [comment]
I now get what's going on here. Automatisch currently only supports `application/json` (in the sense of parsing the request), and it fails in the current release when there is no payload. However, payload-less requests will be covered as of the next release. 

I will leave this issue open until we release the next version.

**Comment by Starttoaster** [comment]
Oh interesting. I also tried it with a data payload, which also caused the main container to crash. But adding the Content-Type header was successful.

`curl -X PUT https://automatisch.my.domain/webhooks/4290d3d8-9470-4a53-af43-c63ba876cd73 -v -d '{"request": "this"}' -H "Content-Type: application/json"`

^ returns a successful response. Omitting the `-H` flag breaks it again :)

**Comment by barinali** [comment]
@Starttoaster, we have just released version 0.4.0. Could you please test the problem and let us know if it's fixed?

**Comment by barinali** [comment]
Closing as completed, as the reported bug has been fixed.
